controllers/subscription_controller.go

```diff
@@ -117,6 +132,72 @@ func (r *SubscriptionReconciler) ensureSubscriptions(logger logr.Logger, namespa
 	return err
 }
 
+func (r *SubscriptionReconciler) setOperatorCondition(logger logr.Logger, namespace string) error {
+	ocdList := &operatorv2.OperatorConditionList{}
+	err := r.Client.List(context.TODO(), ocdList, client.InNamespace(namespace))
+	if err != nil {
+		logger.Error(err, "failed to list OperatorConditions")
+		return err
+	}
+
+	condNames := GetVendorCsvNames(StorageClusterKind)
+	condNames = append(condNames, GetVendorCsvNames(FlashSystemKind)...)
+
+	condMap := make(map[string]struct{}, len(condNames))
+	for i := range condNames {
+		condMap[condNames[i]] = struct{}{}
+	}
+
+	var condNotUpgradeable *metav1.Condition
+	var ocdName string
+	for ocdIdx := range ocdList.Items {
+		// skip operatorconditions of not dependent operators
+		if _, exist := condMap[ocdList.Items[ocdIdx].GetName()]; !exist {
+			continue
+		}
+
+		ocd := &ocdList.Items[ocdIdx]
+		// overrides is of higher priority than operator set conditions
+		cond := util.Find(ocd.Spec.Overrides, func(cd *metav1.Condition) bool {
+			return cd.Type == operatorv2.Upgradeable
+		})
+		if cond != nil {
+			// we found a override for Upgradeable type, if we aren't upgradeable we store the
+			// condition or else break to skip checking Status.Conditions
+			if cond.Status != "True" {
+				condNotUpgradeable = cond
+				ocdName = ocd.GetName()
+				// no need to check another operator
+				break
+			}
+			// we must look at other operator condition overrides but skip checking current operator conditions
+			continue
+		}
+
+		condNotUpgradeable = util.Find(ocd.Status.Conditions, func(cd *metav1.Condition) bool {
+			return cd.Type == operatorv2.Upgradeable && cd.Status != "True"
+		})
+		if condNotUpgradeable != nil {
+			ocdName = ocd.GetName()
+			// no need to check another operator, no override and operator is not upgradeable
+			break
+		}
```

Comment:
Lets create a func for this logic which gets the conditions and check if operator is upgradable or not.

Reply:
I couldn't, I prefer it to be inline, to refactor into a function I need to duplicate some vars which doesn't look apt.

Reply:
done.